addons/product/migrations/12.0.1.2/end-migration.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def update_product_template_attribute_value(env):
+    # the product.template.attribute.values are recomputed each time a
+    # template.attribute.line changes. The conditions to generate this
+    # template_attribute_value have changed, so we need to recompute them.
+    lines = env['product.template.attribute.line'].search([])
+    lines._update_product_template_attribute_values()
```

Comment:
Can we reconstruct only when needed (when different conditions are met)?

Reply:
I will check if we can filter them to just the needed cases.

Reply:
filtered!